libcxx/include/__configuration/abi.h

```diff
@@ -150,6 +176,11 @@
 // ABI impact: changes the iterator type of `vector` (except `vector<bool>`).
 // #define _LIBCPP_ABI_BOUNDED_ITERATORS_IN_VECTOR
 
+// [[msvc::no_unique_address]] seems to mostly affect empty classes, so the padding scheme for Itanium doesn't work.
```

Comment:
Even if it doesn't affect non-empty types, `_LIBCPP_ABI_NO_COMPRESSED_PAIR_PADDING` also removes the alignment adjustment for the compressed pair members on Microsoft ABI. Is that intentional?

Reply:
Yes, we take the ABI break unconditionally on MSVC. It's currently not stable anyways, so we don't need to make our lives harder by trying to figure out how to avoid the ABI break on MSVC.